python/ray/data/_internal/arrow_block.py

```diff
@@ -169,16 +169,20 @@ def __init__(self, table: "pyarrow.Table"):
     def column_names(self) -> List[str]:
         return self._table.column_names
 
-    def append_column(self, name: str, data: Any) -> Block:
+    def fill_column(self, name: str, value: Any) -> Block:
         assert name not in self._table.column_names
 
-        if any(isinstance(item, np.ndarray) for item in data):
-            raise NotImplementedError(
-                f"`{self.__class__.__name__}.append_column()` doesn't support "
-                "array-like data."
-            )
+        import pyarrow as pa
+        import pyarrow.compute as pc
+
+        if isinstance(value, pa.Scalar):
+            type = value.type
+        else:
+            type = pa.infer_type([value])
 
-        return self._table.append_column(name, [data])
+        array = pa.nulls(len(self._table), type=type)
+        array = pc.fill_null(array, value)
```

Comment:
Value have to be converted to Pyarrow `Scalar`

Reply:
What's the advantage of explicitly converting `value` to a scalar rather than letting PyArrow internally handle the conversion?

Reply:
Does it work w/o converting to scalar?